packages/api-client/src/rest/rest.ts

```diff
@@ -124,6 +140,25 @@ export class RestClientImpl implements RestClient {
     return result
   }
 
+  private async checkRateLimits (response: Response): Promise<void> {
+    if (response.status === 429) {
+      // Extract rate limit information from headers
+      const retryAfter = response.headers.get('Retry-After')
+      const rateLimitReset = response.headers.get('X-RateLimit-Reset')
+      // const rateLimitLimit: string | null = response.headers.get('X-RateLimit-Limit')
+      const waitTime =
+        retryAfter != null
+          ? parseInt(retryAfter, 10)
+          : rateLimitReset != null
+            ? new Date(parseInt(rateLimitReset, 10)).getTime() - Date.now()
+            : 1000 // Default to 1 seconds if no headers are provided
+
+      console.warn(`Rate limit exceeded. Waiting ${waitTime / 1000} seconds before retrying...`)
+      await new Promise((resolve) => setTimeout(resolve, waitTime))
+      throw new Error(rateLimitError)
```

Comment:
throw after waiting for timeout? I think we should either throw an error or wait, not both. If we waited for XX seconds, then we can safely do requests again

Reply:
It is used with withRetry and require exception to retry.